File: cpp/estruturas-controle/EstruturaDoWhile.cpp
#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

int main () {

    double C, F;
    char resp;

    do {
        cout << "Digite a temperatura em Celsius: ";
        cin >> C;

        F = 9.0 * C / 5.0 + 32.0;

        cout << fixed << setprecision(1);
        cout << "Equivalente em Fahrenheit: " << F << endl;

        cout << "Deseja repetir? [S/N]: ";
        cin >> resp;

    } while (resp == 's');

    return 0;
}

File: java/estruturas_controle/src/EstruturaDoWhile.java
import java.util.Locale;
import java.util.Scanner;

public class EstruturaDoWhile {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner leitor = new Scanner(System.in);
		
		double F, C;
		char resp;
		
		do {
			System.out.print("Digite a temperatura em Celsius: ");
			C = leitor.nextDouble();
			
			F = 9.0 * C / 5.0 + 32.0;
			
			System.out.println("Equivalente em Fahrenheit: " + String.format("%.1f", F));
			System.out.println("Deseja repetir? [S/N]: ");
			resp = leitor.next().charAt(0);
			
		} while (resp == 's');
		
		leitor.close();
	}
}